app/viewitem.tsx

```diff
@@ -0,0 +1,73 @@
+import React from "react";
+import { View, Text, ActivityIndicator, Button, StyleSheet } from "react-native";
+import { useLocalSearchParams, useRouter } from "expo-router";
+import { useSingleActivity } from "../hooks/useActivity";
+
+const ViewItem = () => {
+    const id = parseInt(useLocalSearchParams<{ activityId: string }>().activityId);
+    const { useFetchActivity } = useSingleActivity({ activityId: id });
+    const { data, error, isLoading } = useFetchActivity;
+    const router = useRouter();
+
+    const handlePress = () => {
+        router.push("/weekplanscreen");
+    }
+
+    if (isLoading) {
+        return <ActivityIndicator size="large" color="#0000ff" />;
+    }
+    if (error) {
+        return <Text style={styles.errorText}>Error fetching activities: {error.message}</Text>;
+    }
+
+    return (
+        <View style={styles.container}>
+            {data && (
+                <View style={styles.activityContainer}>
+                    <Text style={styles.activityName}>{data.name}</Text>
+                    <Text style={styles.activityDescription}>{data.description}</Text>
+                    <Text style={styles.activityTime}>{data.startTime} - {data.endTime}</Text>
+                </View>
+            )}
+            <Button title='Go Back' onPress={handlePress} />
```

Comment:
This is wrong use href

Reply:
Hvis du bare mener at lave the til router.back i stedet så er det gjort